Src/GoogleApis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -99,5 +110,45 @@ public Initializer(string tokenUrl)
         }
 
         #endregion
+
+        /// <summary>
+        /// Return whether code is running on Google Compute Engine.
+        /// </summary>
+        public static async Task<bool> IsRunningOnComputeEngine()
+        {
+            try
+            {
+                Logger.Info("Checking connectivity to ComputeEngine metadata server.");
+                var httpRequest = new HttpRequestMessage(HttpMethod.Get, MetadataServerUrl);
+
+                CancellationTokenSource cts = new CancellationTokenSource();
+                cts.CancelAfter(MetadataServerPingTimeoutInMilliseconds);
+
+                var httpClient = new HttpClientFactory().CreateHttpClient(new CreateHttpClientArgs());

```

Comment:
It's a shame that we can reuse the factory. What if the developer already uses some existing http setting, you are not going to use them.